### PR TITLE
Handlebars: Default --html-whitespace-sensitivity to strict

### PR DESCRIPTION
**Pull request by dcyriller**
## Description

This is a proposal to default `--html-whitespace-sensitivity` option to `strict`. Indeed, this mode is HTML-whitespaces sensitive. As opposed to the current default (`ignore`).

The current default mode would still be supported through the `--html-whitespace-sensitivity ignore` option.

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
Handlebars parser still alpha?  So we don't need a  major release to do this?

**Comment by thorn0** [comment]
@dcyriller Please resolve the conflicts

**Comment by dcyriller** [comment]
The conflicts are resolved. (+ I pushed a fix regarding how the final newline is handled)

**Review by thorn0 — APPROVED** [review]

**Comment by thorn0** [comment]
@dcyriller Would you like to submit another PR to finally move Handlebars support out of alpha to release? Or is there still something that makes this move premature?

**Comment by Alonski** [comment]
@dcyriller @thorn0 Let's do it? The Ember community is starting to adopt the parser even with it being experimental. Might as well make it official? There will always be more bugs

**Comment by thorn0** [comment]
Apart from the necessary code changes, that PR would need a changelog entry with some text (at least a paragraph) about Handlebars support in Prettier. The text will end up in the Highlights section of the release blog post ([example 1](https://prettier.io/blog/2018/11/07/1.15.0.html#highlights), [example 2](https://prettier.io/blog/2017/11/07/1.8.0.html)). Who can write it?

**Comment by dcyriller** [comment]
I have opened https://github.com/prettier/prettier/pull/10290

I'm going to update the changelog with your suggestion @thorn0

**Comment by Alonski** [comment]
So @thorn0 lets merge this in and in #10290 it will get moved out of experimental?

File: tests/handlebars/basics/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -39,37 +39,26 @@ printWidth: 80
...
-</div>
+  /></div>
 
-<Form as |f|>
-  <f.input @title="hello" />
-  <f.input>
-    hello
-  </f.input>
-</Form>
+<Form as |f|><f.input @title="hello" />
```
Comment (dcyriller) [inline tests/handlebars/basics/__snapshots__/jsfmt.spec.js.snap:46]: this needs to be fixed, I think it was introduced in latest rebase

Comment (dcyriller) [reply tests/handlebars/basics/__snapshots__/jsfmt.spec.js.snap]: indeed it was 😅

**Comment by thorn0** [comment]
The tests show there is an idempotence issue:

**Prettier pr-10201**
[Playground link](https://deploy-preview-10201--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4D4BimATgM4wAEAynJFOuQCoAWx9+EArkTE+YQGY8qmAB5Dq2BEICimAOZMhAOUyxeDKb2kAbOJLWMA7nG2D1LbnE19Ol66gD0WPOXIgANCAgAHGJmgkyKAAhkREEIYACqEIgSjB2obBAJ6BngBGRMFgANZwMJTe2apyyDBEHHCecAC26XDo6A0AMsFQchzBcnDsRDXBMH7tyCDBHDAQHiCKNdoA6izwJEVgcNRQJJh+2FvJI2AkaSCqJHDckVly-cj8CaeeAFYkIgBCWbn5lME1cM2qcDc7lUQE8RJQSroAIocCDwQHae4gIqkM4jOTaTA1H5EKbeIiqGBzTDoHjIAAcAAZPHiIKc5llvCM8XBTkRJFMAI4w+AXHxxUYkAC0UCsTXQUyIcC5xDgFy61yQtwRwNONUwZQqKohcGhsIBiqBnhgwXSRJJTGQACYjVlMBj2gBhCBY4IjFkAVimHFODBNcSViOwlQAknQpJQwPjfABBOiUGDJXTw04AXxTQA)
```sh
--parser glimmer
```

**Input:**
```hbs
<div>First Second Third Fourth Fifth Sixth Seventh Eighth Ninth Tenth Eleventh Twelfth Thirteenth Fourteenth </div>  
```

**Output:**
```hbs
<div>First Second Third Fourth Fifth Sixth Seventh Eighth Ninth Tenth Eleventh
  Twelfth Thirteenth Fourteenth
  </div>
```

**Second Output:**
```hbs
<div>First Second Third Fourth Fifth Sixth Seventh Eighth Ninth Tenth Eleventh
  Twelfth Thirteenth Fourteenth
</div>
```

**Pull request merged**